src/videoserver/apps/projects/routes.py

```diff
@@ -1226,16 +1224,13 @@ def _get_preview_thumbnail(self, position, crop, rotate):
                 raise BadRequest({"crop": [{"width": ["crop's frame is outside a video's frame"]}]})
             elif crop['y'] + crop['height'] > self.project['metadata']['height']:
                 raise BadRequest({"crop": [{"height": ["crop's frame is outside a video's frame"]}]})
-
+        # validate position param
+        if self.project['metadata']['duration'] < position:
+              position = self.project['metadata']['duration']
         # resource is busy
```

Comment:
Video duration in client ( check by HTML5) sometime is greater than duration in video server (check by Ffmpeg) some milliseconds. eg: HTML5 18.231 and Ffmpeg 18.229. So it raise 404 error when we editing video.
To avoid it, we should update the `trim end` is equal `video duration` when it greater than video duration.